<a href="https://colab.research.google.com/github/nailah21400/HU-DATA-200/blob/main/MEPS_Assignment%20%231.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# Importing Google Drive Connection (For Colab use only)
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


Research Question: How do insurance coverage, race, and English proficiency affect health outcomes for individuals diagnosed with speech and language disorders?

Primary hypothesis: Individuals with Medicaid or no insurance will report worse health outcomes compared to those with private insurance.

 This project aims to provide the following analysis: Descriptive stats will describe the frequency of speech codes throughout the data set by insurance type, race, and English proficiency and a summary stats of health outcomes; Chi-square test: check relationship between insurance type, race, English proficiency and health outcomes; and ANOVA: compare mean health outcomes by insurance type, race, and English proficiency

## Problem Statement
Research indicates that race, ethnicity, type of insurance, and english proficiency are associated with worse health outcomes, access to healthcare and quality of providers. Little is known about the disparities in health outcomes for individuals with speech, language, and swallowing disorders. This study seeks to investigate potential associations between health outcomes for patients with speech, language, and swallowing disorders based on english proficency, race, and type of insurance.

##Research Question

How do insurance coverage, race, and English proficiency affect health outcomes for individuals diagnosed with speech and language disorders?

**Primary Hypothesis:** Individuals with Medicaid or no insurance will report worse health outcomes compared to those with private insurance.

This project aims to provide the following analyses: Descriptive stats will describe the frequency of speech

1. Descriptive statistics will describe the frequency of speech codes throughout the data set by insurance type, race, and English proficiency and provide a summary of health outcomes.
2.   Chi-square test will see if there is a significant relationship between insurance type, race, English proficiency and health outcomes
3. Analysis of Variance (ANOVA) will compare mean health outcomes by insurance type, race, and English proficiency


## Data Definition
**Medical Expenditure Pannel Survey (MEPS)**

Last Updated: April 14, 2023
https://meps.ahrq.gov/mepsweb/about_meps/survey_back.jsp

The Medical Expenditure Panel Survey (MEPS) is a set of survey questions for individuals, families, and their corresponding medical providers across the United States. Medical employers and insurance company data was also included. This survey was conducted by the Agency for Healthcare Researcha and Quality (AHRQ) and United States Department of Health and Human Services (HHS).

The MEPS collects data on health services used, the frequency of use, and expenditures of the services. The Household Component (HC) gathers interview data from a nationally representative sample of families and indiviudals over a two year period. The Insurance Component (IC) data is on what health care plans  private and public employers offer to their employees. The Medical Provider Component (MCP) is supplemental data on the health care providers, hospitals, physicians, and pharmacies identified by the family and individual participants. The MEPS has data avaialble for 1996-2023. More information about collection methods, background information, and history can be found here: https://meps.ahrq.gov/mepsweb/about_meps/survey_back.jsp
  

## Data Collection
This data was accessed through the HHS-AHRQ MEPS GitHub repository. The repository is accessed directly from Colab using different Python libraries and functions.

The datasets used for this analysis were the following 2022 categories:


*   Full Year Consolidated (fyc22)
*   Medical Conditions (cond 22)



## Examine the Dataset

In [ ]:
##Import the libraries
import numpy as np                  # Scientific Computing
import pandas as pd                 # Data Analysis
import matplotlib.pyplot as plt     # Plotting
import seaborn as sns               # Statistical Data Visualization

# Let's make sure pandas returns all the rows and columns for the dataframe
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

# Force pandas to display full numbers instead of scientific notation
pd.options.display.float_format = '{:.0f}'.format

# Library to suppress warnings
import warnings
warnings.filterwarnings('ignore')

In [ ]:
# Importing Google Drive Connection (For Colab use only)
from google.colab import drive
drive.mount('/content/drive', force_remount=True)

Mounted at /content/drive


In [ ]:
import zipfile
import pandas as pd

#View the Medical Conditions(cond22) data
# Open the zip file
with zipfile.ZipFile("/content/drive/MyDrive/MEPS/cond22.xlsx.zip", 'r') as zip_ref:
# Get the name of the Excel file within the zip
    excel_file_name = zip_ref.namelist()[0]
# Extract the Excel file to memory
    with zip_ref.open(excel_file_name) as extracted_file:
# Read the Excel file using pandas
        path = pd.read_excel(extracted_file)

In [ ]:
#Veiw the first 7 rows of the Medical Conditions data frame
cond22 = pd.DataFrame(path)
cond22.head(7)

DUID  PID    DUPERSID  CONDN        CONDIDX  PANEL  CONDRN  AGEDIAG  \
0  2460002  101  2460002101      1  2460002101001     24       1       25   
1  2460002  101  2460002101      2  2460002101002     24       1       68   
2  2460002  101  2460002101      3  2460002101003     24       1       25   
3  2460002  101  2460002101      4  2460002101004     24       1       50   
4  2460002  101  2460002101      5  2460002101005     24       1       -1   
5  2460002  101  2460002101      6  2460002101006     24       1       -1   
6  2460002  101  2460002101      7  2460002101007     24       1       -1   

   CRND1  CRND2  CRND3  CRND4  CRND5  CRND6  CRND7  CRND8  CRND9  INJURY  \
0     -1     -1     -1     -1     -1     -1      1      0      1       2   
1     -1     -1     -1     -1     -1     -1      0      0      1       2   
2     -1     -1     -1     -1     -1     -1      1      0      1       2   
3     -1     -1     -1     -1     -1     -1      1      0      1       2   
4     -1     -1     -1     -1     -1     -1      0      0      1       2   
5     -1     -1     -1     -1     -1     -1      1      0      1       2   
6     -1     -1     -1     -1     -1     -1      0      0      1       2   

   ACCDNWRK ICD10CDX  CCSR1X  CCSR2X CCSR3X CCSR4X  HHCOND  IPCOND  OPCOND  \
0        -1      I10  CIR007      -1     -1     -1       2       2       2   
1        -1      G45  NVS012      -1     -1     -1       1       2       2   
2        -1      E78  END010      -1     -1     -1       2       2       2   
3        -1      E11  END002  END005     -1     -1       2       2       2   
4        -1      -15  NEO000      -1     -1     -1       1       2       2   
5        -1      C34  NEO022      -1     -1     -1       1       2       2   
6        -1      C85  NEO058      -1     -1     -1       1       2       2   

   OBCOND  ERCOND  RXCOND  PERWT22F  VARSTR  VARPSU  
0       2       2       1      5728    2082       1  
1       2       2       2      5728    2082       1  
2       2       2       1      5728    2082       1  
3       2       2       1      5728    2082       1  
4       2       2       2      5728    2082       1  
5       2       2       2      5728    2082       1  
6       2       2       2      5728    2082       1

In [ ]:
#Veiw the bottom 7 rows of the Medical Conditions
cond22.tail(7)

DUID  PID    DUPERSID  CONDN        CONDIDX  PANEL  CONDRN  AGEDIAG  \
83166  2799700  101  2799700101      3  2799700101003     27       1       65   
83167  2799700  101  2799700101      6  2799700101006     27       2       -1   
83168  2799700  101  2799700101      9  2799700101009     27       2       -1   
83169  2799700  101  2799700101     10  2799700101010     27       2       -1   
83170  2799700  101  2799700101     11  2799700101011     27       3       -1   
83171  2799700  101  2799700101     12  2799700101012     27       3       -1   
83172  2799700  101  2799700101     13  2799700101013     27       3       -1   

       CRND1  CRND2  CRND3  CRND4  CRND5  CRND6  CRND7  CRND8  CRND9  INJURY  \
83166      0      1      1     -1     -1     -1     -1     -1     -1       2   
83167      0      1      0     -1     -1     -1     -1     -1     -1       2   
83168      0      1      0     -1     -1     -1     -1     -1     -1       2   
83169      0      1      1     -1     -1     -1     -1     -1     -1       2   
83170      0      0      1     -1     -1     -1     -1     -1     -1       2   
83171      0      0      1     -1     -1     -1     -1     -1     -1       2   
83172      0      0      1     -1     -1     -1     -1     -1     -1       2   

       ACCDNWRK ICD10CDX  CCSR1X CCSR2X CCSR3X CCSR4X  HHCOND  IPCOND  OPCOND  \
83166        -1      I49  CIR017     -1     -1     -1       2       2       2   
83167        -1      I99  CIR032     -1     -1     -1       2       2       2   
83168        -1      H57  EYE012     -1     -1     -1       2       2       1   
83169        -1      M25  MUS010     -1     -1     -1       2       2       2   
83170        -1      -15  INJ049     -1     -1     -1       2       2       2   
83171        -1      Z13  FAC003     -1     -1     -1       2       2       2   
83172        -1      Z76  FAC012     -1     -1     -1       2       2       2   

       OBCOND  ERCOND  RXCOND  PERWT22F  VARSTR  VARPSU  
83166       2       2       1     11060    2001       3  
83167       1       2       2     11060    2001       3  
83168       2       2       2     11060    2001       3  
83169       2       2       1     11060    2001       3  
83170       2       2       1     11060    2001       3  
83171       2       2       1     11060    2001       3  
83172       2       2       1     11060    2001       3

In [ ]:
# display the dimensions of the data
cond22.shape

(83173, 33)

The Medical Conditions dataframe features 83,173 rows and 33 columns. There are a total of 2,744,709 expect data points in this dataset.

In [ ]:
# Check the basic information about the dataset
cond22.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 83173 entries, 0 to 83172
Data columns (total 33 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   DUID      83173 non-null  int64  
 1   PID       83173 non-null  int64  
 2   DUPERSID  83173 non-null  int64  
 3   CONDN     83173 non-null  int64  
 4   CONDIDX   83173 non-null  int64  
 5   PANEL     83173 non-null  int64  
 6   CONDRN    83173 non-null  int64  
 7   AGEDIAG   83173 non-null  int64  
 8   CRND1     83173 non-null  int64  
 9   CRND2     83173 non-null  int64  
 10  CRND3     83173 non-null  int64  
 11  CRND4     83173 non-null  int64  
 12  CRND5     83173 non-null  int64  
 13  CRND6     83173 non-null  int64  
 14  CRND7     83173 non-null  int64  
 15  CRND8     83173 non-null  int64  
 16  CRND9     83173 non-null  int64  
 17  INJURY    83173 non-null  int64  
 18  ACCDNWRK  83173 non-null  int64  
 19  ICD10CDX  83173 non-null  object 
 20  CCSR1X    83173 non-null  ob

No null or missing values were observed in this dataset. However, the number of non-null values does not macth the expected total of entries. The columns titled: 'DUPERSID', 'ICD10CDX', 'CONDIDX', 'IPCOND', 'OPCOND', 'OBCOND', 'HHCOND', and 'PERWT22F' are crucial for analysis.

There is 1 categorical variable, 'ICD10CDX', and 1 float data type 'PERWT22F'. The other variables are all intergers.





## Data Cleaning


In [ ]:
# Headers
# Create a list of the columns in the dataset
cond22Columns = cond22.columns
cond22Columns

Index(['DUID', 'PID', 'DUPERSID', 'CONDN', 'CONDIDX', 'PANEL', 'CONDRN',
       'AGEDIAG', 'CRND1', 'CRND2', 'CRND3', 'CRND4', 'CRND5', 'CRND6',
       'CRND7', 'CRND8', 'CRND9', 'INJURY', 'ACCDNWRK', 'ICD10CDX', 'CCSR1X',
       'CCSR2X', 'CCSR3X', 'CCSR4X', 'HHCOND', 'IPCOND', 'OPCOND', 'OBCOND',
       'ERCOND', 'RXCOND', 'PERWT22F', 'VARSTR', 'VARPSU'],
      dtype='object')

In [ ]:
# Update the Headers for Syntax Consistency
cond22 = cond22.rename(columns={'DUPERSID':'Participant ID',
                                  'ICD10CDX': 'ICD-10 Code',
                                  'CONDIDX': 'Condition ID',
                                  'IPCOND': 'Inpatient Events',
                                  'OPCOND': 'Outpatient Events',
                                  'OBCOND': 'Office-Based Events',
                                  'HHCOND':'Home Health Events',
                                  'PERWT22F': 'Expenditure Participant Weight'})

# View the new columns and update the variable
# Pass the columns to the variable:
cond22Columns = cond22.columns
# Call the variable to see the contents
cond22Columns

Index(['DUID', 'PID', 'Participant ID', 'CONDN', 'Condition ID', 'PANEL',
       'CONDRN', 'AGEDIAG', 'CRND1', 'CRND2', 'CRND3', 'CRND4', 'CRND5',
       'CRND6', 'CRND7', 'CRND8', 'CRND9', 'INJURY', 'ACCDNWRK', 'ICD-10 Code',
       'CCSR1X', 'CCSR2X', 'CCSR3X', 'CCSR4X', 'Home Health Events',
       'Inpatient Events', 'Outpatient Events', 'Office-Based Events',
       'ERCOND', 'RXCOND', 'Expenditure Participant Weight', 'VARSTR',
       'VARPSU'],
      dtype='object')

In [ ]:
# Missing Values

# Determine the number of missing values
cond22.isnull().sum()

DUID                              0
PID                               0
Participant ID                    0
CONDN                             0
Condition ID                      0
PANEL                             0
CONDRN                            0
AGEDIAG                           0
CRND1                             0
CRND2                             0
CRND3                             0
CRND4                             0
CRND5                             0
CRND6                             0
CRND7                             0
CRND8                             0
CRND9                             0
INJURY                            0
ACCDNWRK                          0
ICD-10 Code                       0
CCSR1X                            0
CCSR2X                            0
CCSR3X                            0
CCSR4X                            0
Home Health Events                0
Inpatient Events                  0
Outpatient Events                 0
Office-Based Events               0
ERCOND                            0
RXCOND                            0
Expenditure Participant Weight    0
VARSTR                            0
VARPSU                            0
dtype: int64

In [ ]:
# create a function to determine the percentage of missing values

def missing(DataFrame):
    print ('Percentage of missing values in the dataset:\n',
           round((DataFrame.isnull().sum() * 100/ len(DataFrame)),2).sort_values(ascending=False))


# Call the function and execute
missing(cond22)

Percentage of missing values in the dataset:
 DUID                             0
PID                              0
Participant ID                   0
CONDN                            0
Condition ID                     0
PANEL                            0
CONDRN                           0
AGEDIAG                          0
CRND1                            0
CRND2                            0
CRND3                            0
CRND4                            0
CRND5                            0
CRND6                            0
CRND7                            0
CRND8                            0
CRND9                            0
INJURY                           0
ACCDNWRK                         0
ICD-10 Code                      0
CCSR1X                           0
CCSR2X                           0
CCSR3X                           0
CCSR4X                           0
Home Health Events               0
Inpatient Events                 0
Outpatient Events                0
Office-Ba

There are no missing values in the Medical Conditions dataset.

In [ ]:
# Make a copy of the DataFrame before manipulation
cond22OG = cond22

In [ ]:
# Dropping columns from the dataframe
cond22 = cond22.drop(['DUID', 'PID','CONDN', 'PANEL', 'CONDRN','AGEDIAG' , 'CRND1', 'CRND2', 'CRND3', 'CRND4', 'CRND5', 'CRND6', 'CRND7', 'CRND8', 'CRND9', 'INJURY', 'ACCDNWRK','CCSR1X', 'CCSR2X', 'CCSR3X', 'CCSR4X','ERCOND', 'RXCOND','VARSTR', 'VARPSU' ],
            axis=1)

# Check the .info() for results
cond22.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 83173 entries, 0 to 83172
Data columns (total 8 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Participant ID                  83173 non-null  int64  
 1   Condition ID                    83173 non-null  int64  
 2   ICD-10 Code                     83173 non-null  object 
 3   Home Health Events              83173 non-null  int64  
 4   Inpatient Events                83173 non-null  int64  
 5   Outpatient Events               83173 non-null  int64  
 6   Office-Based Events             83173 non-null  int64  
 7   Expenditure Participant Weight  83173 non-null  float64
dtypes: float64(1), int64(6), object(1)
memory usage: 5.1+ MB
